components/prize_pool/commons/prize_pool_placement_base.lua

```diff
@@ -0,0 +1,189 @@
+			opponent.additionalData = self:readAdditionalData(opponentInput)
+
+			-- Set date
+			opponent.date = opponentInput.date
+		end
+		return opponent
+	end)
+end
+
+function Placement:_shouldAddTbdOpponent(opponentIndex, place)
+	-- We want at least 1 opponent present for all placements
+	if opponentIndex == 1 then
+		return true
+	end
+	-- If the fillPlaceRange option is disabled or we do not have a give placeRange do not fill up further
+	if not self.parent.options.fillPlaceRange or not self.count then
+		return false
+	end
+	-- Only fill up further with TBD's if there is free space in the placeRange/slot
+	if opponentIndex <= self.count then
+		return true
+	end
+	return false
+end
+
+function Placement:readAdditionalData(args)
+	error('Function readAdditionalData needs to be set via the module that requires "Module:PrizePool/Placement/Base"')
+end
+
+function Placement:parseOpponentArgs(input, date)
+	-- Allow for lua-table, json-table and just raw string input
+	local opponentArgs = Json.parseIfTable(input) or (type(input) == 'table' and input or {input})
+	opponentArgs.type = opponentArgs.type or self.parent.opponentType
+	assert(Opponent.isType(opponentArgs.type), 'Invalid type')
+
+	local opponentData
+	if type(opponentArgs[1]) == 'table' and opponentArgs[1].isAlreadyParsed then
+		opponentData = opponentArgs[1]
+	elseif type(opponentArgs[1]) ~= 'table' then
+		opponentData = Opponent.readOpponentArgs(opponentArgs)
+	end
+
+	if not opponentData or (Opponent.isTbd(opponentData) and opponentData.type ~= Opponent.literal) then
+		opponentData = Table.deepMergeInto(Opponent.tbd(opponentArgs.type), opponentData or {})
+	end
+
+	return Opponent.resolve(opponentData, date, {syncPlayer = self.parent.options.syncPlayers})
+end
+
+
+function Placement:getPrizeRewardForOpponent(opponent, prize)
+	return opponent.prizeRewards[prize] or self.prizeRewards[prize]
+end
+
+function Placement:_setBaseFromRewards(prizesToUse, prizeTypes)
+	Array.forEach(self.opponents, function(opponent)
+		if opponent.prizeRewards[PRIZE_TYPE_BASE_CURRENCY .. 1] or self.prizeRewards[PRIZE_TYPE_BASE_CURRENCY .. 1] then
+			return
+		end
+
+		local baseReward = 0
+		Array.forEach(prizesToUse, function(prize)
+			local localMoney = opponent.prizeRewards[prize.id] or self.prizeRewards[prize.id]
+
+			if not localMoney or localMoney <= 0 then
+				return
+			end
+
+			baseReward = baseReward + prizeTypes[prize.type].convertToBaseCurrency(
+				prize.data,
+				localMoney,
+				opponent.date,
+				self.parent.options.currencyRatePerOpponent
+			)
+			self.parent.usedAutoConvertedCurrency = true
+		end)
+
+		opponent.prizeRewards[PRIZE_TYPE_BASE_CURRENCY .. 1] = baseReward
+	end)
+end
+
+--- Returns true if the input matches the format of a date
+function Placement._isValidDateFormat(date)
+	if type(date) ~= 'string' or String.isEmpty(date) then
+		return false
+	end
+	return date:match('%d%d%d%d%-%d%d%-%d%d') and true or false
+end
```

Comment:
Do we have something that does this somewhere in the standard lib?

Reply:
nothing that just checks it and returns a bool afaik
(atm it is just a plain (ctrl+x ctrl+v) move from /Placement to /Placement/Base)